src/contracts.ts

```diff
@@ -1,11 +1,16 @@
 import { PartialReadonlyContractAbi } from './types'
+import { AbiType, MethodAbi, EventAbi } from 'web3'
 
-export const getMethod = (abi: PartialReadonlyContractAbi, name: string) => {
-  return abi.filter(x => x.type == 'function' && x.name == name)[0]
+export const getMethod = (abi: PartialReadonlyContractAbi, name: string): MethodAbi => {
+  // Have to cast since there's a bug in
```

Comment:
It's not a bug. It's impossible to tell from the definition of `PartialReadonlyContractAbi` that the type should resolve to `MethodAbi`. To fix you need to properly type `PartialReadonlyContractAbi`.

Reply:
@alexanderatallah want to make sure this suggestion didn't get lost